runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/stream/GrpcClientFactory.java

```diff
@@ -450,21 +465,49 @@ private void doAppData(
             assert replySeq <= replyAck + replyMax;
         }
 
-        private void doAppAbort(
+        private void doAppAbortDeferring(
+            long traceId,
+            long authorization,
+            String16FW grpcStatus)
+        {
+            this.grpcStatus = grpcStatus != null ? grpcStatus.asString() : null;
+            this.state = GrpcState.closingReply(state);
+
+            if (GrpcState.replyOpened(state))
+            {
+                doAppAbortDeferred(traceId, authorization);
+            }
+        }
+
+        private void doAppAbortDeferred(
             long traceId,
             long authorization)
         {
-            if (!GrpcState.replyClosed(state))
+            GrpcAbortExFW abortEx = grpcStatus != null
+                    ? grpcAbortExRW.wrap(extBuffer, 0, extBuffer.capacity())
+                            .typeId(grpcTypeId)
+                            .status(grpcStatus)
+                            .build()
+                    : grpcAbortExRW.wrap(extBuffer, 0, extBuffer.capacity())
+                            .typeId(grpcTypeId)
+                            .status(HEADER_VALUE_GRPC_INTERNAL_ERROR)
+                            .build();
```

Comment:
Seems like the only diff is status. Wouldn't it be shorter if we just hoist that value only?

Reply:
Right, but they are different types so that doesn't work. 

We need to have the grpcStatus live longer than a method call, so we keep it as a String, whereas the common case precomputes the value, not as a String.

Reply:
Oh I see I missed that thanks